File: 3-Lab-Solutions/Sample-Microservices/dotnet/Xerris.SRE.ProductCatalog/src/Xerris.SRE.ProductCatalog/Models/Product.cs
namespace Xerris.SRE.ProductCatalog.Models
{
    public class Product
    {
        public Guid Sku { get; set; }
        public string Name { get; set; } = null;
        public string Description { get; set; }
        public string Manufacturer { get; set; }
        public Price Price { get; set; }

        public Product SetPrice(Price pr)
        {
            Price = pr;
            return this;
        }
    }
}

File: Sample-Microservices/Java/Product Catalog/src/main/java/com/xerris/sre/products/domain/Product.java
package com.xerris.sre.products.domain;

import java.util.UUID;

public class Product {

    public UUID getSku() { return sku; }
    public void setSku(UUID sku) {this.sku = sku; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getManufacturer() { return manufacturer; }
    public void setManufacturer(String manufacturer) { this.manufacturer = manufacturer; }

    public Price getPrice() { return price; }
    public void setPrice(Price price) { this.price = price; }

    public Product(UUID sku, String name, String description, String manufacturer) {
        this.sku = sku;
        this.name = name;
        this.description = description;
        this.manufacturer = manufacturer;
    }

    private UUID sku;
    private String name;
    private String description;
    private String manufacturer;
    private Price price;
}

